dart/dynamics/ReferentialSkeleton.hpp

```diff
@@ -70,6 +80,15 @@ class ReferentialSkeleton : public MetaSkeleton
   /// \{ \name Structural Properties
   //----------------------------------------------------------------------------
 
+  /// Returns number of skeletons associated with this ReferentialSkeleton.
+  std::size_t getNumSkeletons() const;
+
+  /// Returns skeletons associated with this ReferentialSkeleton.
+  const std::unordered_set<const Skeleton*>& getSkeletons() const;
```

Comment:
I feel kind of the same way about this function as well, in terms of needlessly exposing implementation detail.

As I understand it, the goal of the `ReferentialSkeleton` class is to abstract away the existence of individual `Skeleton` objects. I would suggest that we remain non-committal about the underlying storage/management of those Skeletons, and not expose these kinds of functions to users.

That said, I'm not against `getNumSkeletons()` or `hasSkeleton(~)` since those don't expose implementation.

Reply:
I guess it's really `MetaSkeleton` that exists to abstract away the individual Skeletons. Still, I think we should avoid exposing implementation to avoid committing ourselves to a particular implementation.

Reply:
I agree. I would love to return a type-erased range here, but I don't know of a good way of doing that in any existing version of C++. The best option we have to today would be to return separate `begin()` and `end()` iterators, but that would be pretty gross to use as a consumer of the API.

Reply:
Removed.

@mkoval I'm thinking it would be good to provide iterator APIs for other classes whenever relevant, but let's save it for future changes. :sweat_smile: